src/chain.h

```diff
@@ -462,7 +462,7 @@ class CChain
     /** Return the maximal height in the chain. Is equal to chain.Tip() ? chain.Tip()->nHeight : -1. */
     int Height() const
     {
-        return vChain.size() - 1;
+        return int(vChain.size()) - 1;
```

Comment:
What is the advantage of this approach instead of static casts? i.e. if braced initialization is used the compiler suggests the latter.
```
chain.cpp:154:27: error: non-constant-expression cannot be narrowed from type 'uint64_t' (aka 'unsigned long') to 'int64_t' (aka 'long') in initializer list [-Wc++11-narrowing]
    return sign * int64_t{r.GetLow64()};
                          ^~~~~~~~~~~~
chain.cpp:154:27: note: insert an explicit cast to silence this issue
    return sign * int64_t{r.GetLow64()};
                          ^~~~~~~~~~~~
                          static_cast<int64_t>( )
1 error generated.

.../...

./chain.h:478:20: error: non-constant-expression cannot be narrowed from type 'std::vector::size_type' (aka 'unsigned long') to 'int' in initializer list [-Wc++11-narrowing]
        return int{vChain.size()} - 1;
                   ^~~~~~~~~~~~~
./chain.h:478:20: note: insert an explicit cast to silence this issue
        return int{vChain.size()} - 1;
                   ^~~~~~~~~~~~~
                   static_cast<int>( )
1 error generated.
```

Reply:
The two give the same result, but one of them is less typing and bloat.